.evergreen/verify-artifacts.sh

```diff
@@ -0,0 +1,65 @@
+#! /usr/bin/env bash
+
+set -e
+
+ARTIFACTS_DIR="packages/compass/dist"
+echo "Verifying artifacts at $ARTIFACTS_DIR"
+ls -l $ARTIFACTS_DIR
+
+TMP_FILE=$(mktemp)
+trap_handler() {
+  local code=$?
+  if [ $code -eq 0 ]; then
+    echo "Verification successful"
+  else
+    echo "Verification failed with exit code $code"
+    cat "$TMP_FILE"
+  fi
+  rm -f "$TMP_FILE"
+  exit $code
+}
+
+trap trap_handler ERR EXIT
+
+verify_using_gpg() {
+  echo "Verifying $1 using gpg"
+  gpg --verify $ARTIFACTS_DIR/$1.sig $ARTIFACTS_DIR/$1 > "$TMP_FILE" 2>&1
+}
+
+verify_using_powershell() {
+  echo "Verifying $1 using powershell"
+  powershell Get-AuthenticodeSignature -FilePath $ARTIFACTS_DIR/$1 > "$TMP_FILE" 2>&1
+}
+
+verify_using_codesign() {
+  echo "Verifying $1 using codesign"
+  codesign -dv --verbose=4 $ARTIFACTS_DIR/$1 > "$TMP_FILE" 2>&1
+}
+
+# For Windows
+if [ "$IS_WINDOWS" = true ]; then
+  verify_using_powershell $WINDOWS_EXE_NAME
+  verify_using_powershell $WINDOWS_MSI_NAME
+
+  echo "Skipping verification for Windows artifacts using gpg: $WINDOWS_ZIP_NAME, $WINDOWS_NUPKG_NAME"
+  exit 0
+fi
+
+
+echo "Importing Compass public key"
+curl https://pgp.mongodb.com/compass.asc | gpg --import > "$TMP_FILE" 2>&1
```

Comment:
This modifies global state on the machine on which it runs – that's probably fine most of the time, but it's not ideal, and it's probably fairly easy to work around it by passing `--homedir` to all GPG invocations with some sort of temporary directory (a new one created for the script or even just $PWD)?

Reply:
added in [ef6c717](https://github.com/mongodb-js/compass/pull/5365/commits/ef6c717d2ff509b48e1bb874520c42cc21130d52)